src/main/java/us/ihmc/commons/thread/RepeatingTaskThread.java

```diff
@@ -0,0 +1,390 @@
+package us.ihmc.commons.thread;
+
+import us.ihmc.commons.Conversions;
+import us.ihmc.commons.RunnableThatThrows;
+import us.ihmc.commons.exception.DefaultExceptionHandler;
+import us.ihmc.commons.exception.ExceptionHandler;
+import us.ihmc.commons.exception.ExceptionTools;
+
+/**
+ * A thread that executes code in a loop.
```

Comment:
A thread that repeats execution of a single task. It can do this N times, continuously, or at a constant rate.